cfme/tests/distributed/test_appliance_replication.py

```diff
@@ -275,3 +281,54 @@ def test_replication_connect_to_vm_in_region(provider,
         view.login.click()
         view = vm.create_view(InfraVmDetailsView)
         wait_for(lambda: view.is_displayed, message="Wait for VM Details page")
+
+
+@pytest.mark.ignore_stream("upstream")
+def test_appliance_httpd_roles(temp_appliance_preconfig_funcscope_rhevm,
+        temp_appliance_unconfig_funcscope_rhevm):
+    """Test that a secondary appliance only runs httpd if a server role requires it.
+    Disable all server roles that require httpd, and verify that httpd is stopped. For each server
+    role that requires httpd, enable it (with all other httpd server roles disabled), and verify
+    that httpd starts.
+
+    Bugzilla:
+        1449766
+
+    Metadata:
+        test_flag: configuration
+
+    Polarion:
+        assignee: tpapaioa
+        casecomponent: Appliance
+        caseimportance: medium
+        initialEstimate: 1/4h
+    """
+    primary_appliance = temp_appliance_preconfig_funcscope_rhevm
+    secondary_appliance = temp_appliance_unconfig_funcscope_rhevm
+    configure_distributed_appliances(primary_appliance, secondary_appliance)
+
+    roles = HTTPD_ROLES.pick(secondary_appliance.version)
+    fill_values = {k: False for k in roles}
+
+    # Change roles through primary appliance to guarantee UI availability.
+    sid = secondary_appliance.server.sid
+    secondary_server = primary_appliance.collections.servers.instantiate(sid=sid)
+
+    primary_appliance.browser_steal = True
+    with primary_appliance:
+        view = navigate_to(secondary_server, 'Server')
+
+        for role in roles:
+            # Disable all httpd roles and verify that httpd is stopped.
+            view.server_roles.fill(fill_values)
```

Comment:
This is a clever use of WT - it will only update the fields that need to be updated in the fill loop.

So each time through, one role gets enabled and tested, and then at the start of the loop here only that enabled role will get disabled. All the roles that are already disabled will just be left alone in the form fill.